### DISCUSSION TITLE
Styles gone after applying tokens

### DISCUSSION THREAD

**Discussion by LeJanusz** [six7/figma-tokens#443] [Q&A]
Steps to recreate:
1)Tokens created in Library File
2) Styles created in Library File

3) Styles applied in a new file.
4) Changing token set. 
5) All styles are replaced with the hex value of the selected token set.

Is this a bug, limitation or did I do sth wrong along the way? :)

//Styles are not replaced in the library file itself.

**Comment by six7** [comment]
This happens as the plugin doesn't save figmaStyleIds on each style. To apply a style, we'd need to know what ID a style has we want to apply, right now all we store is a token name.

However, Color styles are Figma's version of tokens, so one of my workarounds for that is to use Color styles wherever possible and not apply color tokens via the plugin. Instead I use the plugin to create and manage those styles for me.

https://jansix.at/blog/creating-design-systems-based-on-design-tokens-figma-tokens

In the future we want to tackle this to allow theme swaps as you described while keeping styles in place. Here's an issue for that if you want to follow along: https://github.com/six7/figma-tokens/issues/145
